File: python/svix/api_internal/endpoint_auto_config.py
# This file is @generated
from ..models import EndpointOut, SubscribeIn
from .common import ApiBase


class EndpointAutoConfigAsync(ApiBase):
    async def update(
        self, app_id: str, endpoint_id: str, subscribe_in: SubscribeIn
    ) -> EndpointOut:
        """Update an auto-config endpoint by providing endpoint details."""
        response = await self._request_asyncio(
            method="put",
            path="/api/v1/app/{app_id}/endpoint/{endpoint_id}/auto-config",
            path_params={
                "app_id": app_id,
                "endpoint_id": endpoint_id,
            },
            json_body=subscribe_in.model_dump_json(exclude_unset=True, by_alias=True),
        )
        return EndpointOut.model_validate(response.json())


class EndpointAutoConfig(ApiBase):
    def update(
        self, app_id: str, endpoint_id: str, subscribe_in: SubscribeIn
    ) -> EndpointOut:
        """Update an auto-config endpoint by providing endpoint details."""
        response = self._request_sync(
            method="put",
            path="/api/v1/app/{app_id}/endpoint/{endpoint_id}/auto-config",
            path_params={
                "app_id": app_id,
                "endpoint_id": endpoint_id,
            },
            json_body=subscribe_in.model_dump_json(exclude_unset=True, by_alias=True),
        )
        return EndpointOut.model_validate(response.json())

File: ruby/lib/svix/api_internal/endpoint_autoconfig.rb
# frozen_string_literal: true
# This file is @generated

require "net/http"

module Svix
  module ApiInternal
    class EndpointAutoconfig
      def initialize(client)
        @client = client
      end

      def get(app_id, autoconfig_id)
        res = @client.execute_request(
          "GET",
          "/api/v1/app/#{app_id}/autoconfig/#{autoconfig_id}"
        )
        AutoConfigSubscriptionOut.deserialize(res)
      end

      def subscribe(app_id, autoconfig_id, endpoint_in)
        res = @client.execute_request(
          "PUT",
          "/api/v1/app/#{app_id}/autoconfig/#{autoconfig_id}/endpoint",
          body: endpoint_in
        )
        EndpointOut.deserialize(res)
      end

    end
  end
end
